src/functions-and-arrays.js

```diff
@@ -1,38 +1,109 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(num1, num2) {
+  if (num1>num2){
+    return num1;
+  } else {
+    return num2;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  if (words.length===0){
+    return null;
+  }
+ let longestWord = words[0]
+
+  for (let i = 0; i < words.length; i++) {
+    const currentWord = words[i]
+    if (currentWord.length > longestWord.length) {
+      longestWord = currentWord
+    }
+  }
+  return longestWord
 
+}
+const longestWord = findLongestWord(words);
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let sum = 0;
+  for (let i = 0; i < numbers.length; i++) {
+    sum += numbers[i];
+  }
+  return sum;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
+function sum(arr) {
+  let totalSum = 0;
+
+  for (let i = 0; i < arr.length; i++) {
+    const element = arr[i];
+
+    if (typeof element === 'number' || !isNaN(parseFloat(element))) {
+      totalSum += Number(element);
+    } else if (typeof element === 'string') {
+      totalSum += element.length;
+    } else if (typeof element === 'boolean') {
+      totalSum += element ? 1 : 0;
+    }
```

Comment:
```suggestion
    if (typeof element === 'number' || !isNaN(parseFloat(element))) {
      totalSum += Number(element);
    } else if (typeof element === 'string') {
      totalSum += element.length;
    } else if (typeof element === 'boolean') {
      totalSum += element ? 1 : 0;
    } else {
      throw new Error("Unsupported data type sir or ma'am")
    }
```
De esta forma tendrias resuelto lo que te pedian de lanzar un error pero vamos genial!!